src/wizards/tokenSwapDealWizard/stages/termsStage/termsStage.ts

```diff
@@ -29,8 +29,9 @@ export class TermsStage implements IBaseWizardStage {
     this.stageMetadata = stageMeta.settings;
     this.stageMetadata.termsViewModes = this.stageMetadata.termsViewModes ?? this.getDefaultTermsViewModes(stageMeta.wizardType);
 
+    this.addIdsToClauses(stageMeta.wizardType);
+
     this.wizardService.registerStageValidateFunction(this.wizardManager, async () => {
-      this.addIdsToClauses();
       this.checkedForUnsavedChanges();
```

Comment:
My fix could be debatable.
Since generating the ids is not costly, the "After" approach is "more readable" code wise.

**Before:** Only generate ids when either
- proceeding
- clicking on the step in the stepper

**After: **
- generate ids when the stage is `activated`

The other approach would have been.
1. keep `this.addIdsToClauses()` in line 33.
2. In line 32 add the if check for make-an-offer type

- This would be 2 places where `this.addIdsToClauses` would be called